tools/c7n_azure/c7n_azure/actions/tagging.py

```diff
@@ -281,21 +287,27 @@ def _get_tag_value_from_event(self, event):
         elif StringUtils.equal(principal_type, 'ServicePrincipal'):
             user = self.sp_jmes_path.search(event)
 
-        # Other types and main fallback (e.g. User, Office 365 Groups, and Security Groups)
-        if not user and self.upn_jmes_path.search(event):
-            user = self.upn_jmes_path.search(event)
+        if not user:
```

Comment:
We should add more unit tests that walk through this fallback logic, where if one default claim is missing we try the other.

Reply:
Good point, thanks!